dashboard/lib/pd/survey_pipeline/mapper.rb

```
@@ -71,15 +72,21 @@ def map_to_reducers(groups)
           next unless condition.call(group_key)
 
           reducers.each do |reducer|
-            reducer_result = reducer.reduce group_records.pluck(field)
+            # Only process values that are not nil
+            reducer_result = reducer.reduce group_records.pluck(field).compact
             next unless reducer_result.present?
 
             summaries << group_key.merge({reducer: reducer.name, reducer_result: reducer_result})
+          rescue => e
+            errors << e.message
```

Not blocking on this PR, but I'm interested in a discussion about how specific we can usefully get with this (and similar) `rescue` blocks.  I know we want to be resilient against failures, but I wonder if we can define useful boundaries of that resilience that will help us notice _very_ unexpected scenarios faster.

Related: [Rescue StandardError, Not Exception - ThoughtBot](https://thoughtbot.com/blog/rescue-standarderror-not-exception) (especially the "Best-case Scenario" example)

---

My train of thought when dealing with errors (StandardError) is
1. What code is more likely to produce errors? (Any code could potentially fail, but some has higher risk than the other.)
2. What errors are expected (everything else is unexpected)? 
3. What errors are fatal at what layers? (An error could be fatal and halt execution of a callee function but at the same time is non-fatal to the caller.)
4. What to do when catch an expected/unexpected fatal/non-fatal error? A few options are 
a) Interrupt the current code flow, throw error up 1 layer and let the upper layer deal with it.
b) Swallow the error and continue with the current code flow.
c) Swallow the error but report it via Honeybadger (externally) or record it and bubble the info up to upper layers.

In this particular case, the rescued block executes one reducer function on one group of data, which is a part of applying multiple reducers to multiple groups. I chose option (c) here, swallow all errors, record and bubble them up. This is best-effort approach, getting usable results but still notify higher layers about caught errors.